python/lsst/afw/math/spatialCell.cc

```diff
@@ -20,26 +20,129 @@
  * see <https://www.lsstcorp.org/LegalNotices/>.
  */
 
+#include <memory>
+#include <vector>
+
 #include <pybind11/pybind11.h>
-#include <pybind11/operators.h>
 #include <pybind11/stl.h>
 
 #include "lsst/afw/geom/Box.h"
-
 #include "lsst/afw/math/SpatialCell.h"
 
 namespace py = pybind11;
 using namespace pybind11::literals;
 
-using namespace lsst::afw::math;
-
-PYBIND11_DECLARE_HOLDER_TYPE(MyType, std::shared_ptr<MyType>);
-
-void declareTestClasses(py::module &mod){
+namespace lsst {
+namespace afw {
+namespace math {
+
+namespace {
+
+    using CandidateList = std::vector<std::shared_ptr<SpatialCellCandidate>>;
+
+    // Wrap SpatialCellCandidate (an abstract class so no constructor is wrapped)
+    void wrapSpatialCellCandidate(py::module & mod) {
+        py::class_<SpatialCellCandidate, std::shared_ptr<SpatialCellCandidate>>
+            cls(mod, "SpatialCellCandidate");
+
+        py::enum_<SpatialCellCandidate::Status>(cls, "Status")
+            .value("BAD", SpatialCellCandidate::Status::BAD)
+            .value("GOOD",SpatialCellCandidate::Status::GOOD)
+            .value("UNKNOWN",SpatialCellCandidate::Status::UNKNOWN)
+            .export_values();
+
+        cls.def("getXCenter", &SpatialCellCandidate::getXCenter);
+        cls.def("getYCenter", &SpatialCellCandidate::getYCenter);
+        cls.def("setStatus", &SpatialCellCandidate::setStatus);
+        cls.def("getId", &SpatialCellCandidate::getId);
+    }
+
+    // Wrap SpatialCellCandidateIterator
+    void wrapSpatialCellCandidateIterator(py::module & mod) {
+        py::class_<SpatialCellCandidateIterator> cls(mod, "SpatialCellCandidateIterator");
+        cls.def("__incr__", &SpatialCellCandidateIterator::operator++, py::is_operator());
+        cls.def("__deref__",
+                [](SpatialCellCandidateIterator &it) -> std::shared_ptr<SpatialCellCandidate> { return *it; },
```

Comment:
I agree, but this code was moved from elsewhere. I guessed that automatic deduction would give the wrong type in this case, but I did not test it to see.